aten/src/ATen/native/TensorShape.cpp

```diff
@@ -1693,4 +1693,46 @@ Tensor& diag_cpu_out(Tensor &result, const Tensor& self, int64_t dimension) {
   return result;
 }
 
+Tensor moveaxis(const Tensor& self, IntArrayRef src, IntArrayRef dst) {
+  TORCH_CHECK(src.size() == dst.size(), "moveaxis: Invalid source or destination dims: src (",
+              src.size(), " dims ) should contain the same number of dims as dst (", dst.size(), "dims)");
+  auto normalized_src = src.vec();
+  maybe_wrap_dims(normalized_src, self.dim());
+  auto normalized_dst = dst.vec();
+  maybe_wrap_dims(normalized_dst, self.dim());
+
+  auto it_src = std::unique(normalized_src.begin(), normalized_src.end());
+  TORCH_CHECK(it_src == normalized_src.end(), "moveaxis: repeated axis in `src` (", src, ")");
+  auto it_dst = std::unique(normalized_dst.begin(), normalized_dst.end());
+  TORCH_CHECK(it_dst == normalized_dst.end(), "moveaxis: repeated axis in `dst` (", dst, ")");
```

Comment:
We should probably use `dim` instead of axis even if we call the operator `moveaxis`. So "moveaxis: repeated dim in `dst`". This matches what the `torch.permute` error message says if there aren't unique entries.

Reply:
Done.